internal/projection/process/event_enricher_test.go

```diff
@@ -299,11 +299,32 @@ var _ = Describe("Process message", func() {
 			assertHostsSummary(enrichedEvent)
 		})
 	})
+	When("Enrich a cluster event with changing version metadata", func() {
+		It("keeps compatible output", func() {
+			myMetadata := map[string]interface{}{
+				"versions": map[string]interface{}{
+					"release_tag": "foobar",
+					"versions":    `{"assisted-installer": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-rhel8:latest", "assisted-installer-controller": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-reporter-rhel8:latest", "assisted-installer-service": "quay.io/app-sre/assisted-service:880ad10", "discovery-agent": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:latest"}`,
+				},
+			}
+			message := "Cluster with ID myid updated"
+			event := getEvent("cluster_updated", message, myMetadata)
+			cluster := map[string]interface{}{
+				"foo": "bar",
+			}
+			hosts := getHosts()
+			infraEnvs := getInfraEnvs()
+			enrichedEvent := enricher.GetEnrichedEvent(event, cluster, hosts, infraEnvs)
+			Expect(enrichedEvent.ReleaseTag).To(Equal("foobar"))
+		})
+	})
 })
 
-func getEvent(name, message string) *types.Event {
-	metadata := map[string]interface{}{
-		"foo": "bar",
+func getEvent(name, message string, metadata map[string]interface{}) *types.Event {
```

Comment:
I think it would read better if we have 2 methods: `getEvent` and `getEventWithMetadata`. This way we don't have to pass nill everywhere and complicate logic of this helper method